### ISSUE TITLE
[WordCloud template] Words position not synchronised with corresponding data objects

### ISSUE THREAD

**Issue by IevgenySp** [ecomfe/echarts#2694]
Steps to reproduce:

1) Open wordCloud template example from your site: https://ecomfe.github.io/echarts/doc/example/wordCloud.html#-en
2) Replace code with next example: 

```
function createRandomItemStyle() {
    return {
        normal: {
            color: 'rgb(' + [
                Math.round(Math.random() * 160),
                Math.round(Math.random() * 160),
                Math.round(Math.random() * 160)
            ].join(',') + ')'
        }
    };
}

option = {
    title: {
        text: 'Google Trends',
        link: 'http://www.google.com/trends/hottrends'
    },
    tooltip: {
        show: true
    },
    series: [{
        name: 'Google Trends',
        type: 'wordCloud',
        center: ['50%', '50%'],
        size: ['100%', '100%'],
        textRotation : [0, 45, 90, -45],
        textPadding: 5,
        autoSize: {
            enable: true,
            minSize: 12
        },
        data: [
            {
                name: "Fletcher",
                value: 14958.980000000001,
                itemStyle: {
                    normal: {
                        color: 'black'
                    }
                }
            },
            {
                name: "Phillip",
                value: 6336.9,
                itemStyle: createRandomItemStyle()
            },
            {
                name:"Beau",
                value: 7436.74,
                itemStyle: createRandomItemStyle()
            },
            {
                name: "Carson",
                value: 7222.46,
                itemStyle: createRandomItemStyle()
            },
            {
                name: "Reese",
                value: 8948.939999999999,
                itemStyle: createRandomItemStyle()
            },
            {
                name: "Octavius",
                value: 7527.81,
                itemStyle: createRandomItemStyle()
            },
            {
                name: "Thor",
                value: 4431.69,
                itemStyle: createRandomItemStyle()
            },
            {
                name: "Chester",
                value: 4447.58,
                itemStyle: createRandomItemStyle()
            },
            {
                name: "Boris",
                value: 3777.72,
                itemStyle: createRandomItemStyle()
            },
            {
                name: "Xavier",
                value: 3800.88,
                itemStyle: createRandomItemStyle()
            },
            {
                name: "Oleg",
                value: 4100.09,
                itemStyle: createRandomItemStyle()
            },
            {
                name: "Samuel",
                value: 3002.21,
                itemStyle: createRandomItemStyle()
            },
            {
                name: "Aladdin",
                value: 4128.84,
                itemStyle: createRandomItemStyle()
            },
            {
                name: "Arthur",
                value:  4000.58,
                itemStyle: createRandomItemStyle()
            },
            {
                name: "Dominic",
                value: 1748.94,
                itemStyle: createRandomItemStyle()
            },
            {
                name: "Ira",
                value: 4548.77,
                itemStyle: createRandomItemStyle()
            },
            {
                name: "Hoyt",
                value: 2242.29,
                itemStyle: createRandomItemStyle()
            },
            {
                name: "Owen",
                value: 3008.81,
                itemStyle: createRandomItemStyle()
            },
            {
                name: "Cooper",
                value: 3336.87,
                itemStyle: createRandomItemStyle()
            },
            {
                name: "Alfonso",
                value: 4453.27,
                itemStyle: createRandomItemStyle()
            }
        ]
    }]
};
```

3) Hover over words and notice that tooltip data not correspond to actual words.

Expected result: Hover event callback should return correct data object that correspond to selected word.

P.S. It looks like desync between Data objects and actual visual data representation happens on some step of rendering. Data objects remain the same but position of words on canvas changes and this leeds to labels and actual words mismatch.

**Comment by suxuekun** [comment]
2.27 right ? just wait 3.0 wordcloud ba!

this issue reported long long ago.

https://github.com/ecomfe/echarts/issues/1967

**Comment by IevgenySp** [comment]
@suxuekun Thanks for quick answer. Do you know an approximate date when new WordCloud will be released? Also will new template be compatible with ECharts 2.0 or ECharts 3.0 architecture is completely different?

**Comment by heremiguel** [comment]
@suxuekun also waiting for this fix. Thank you very much.

**Comment by pissang** [comment]
Hi @heremiguel @IevgenySp. The problem of wordCloud has already been fixed in the source code. But since we have frozen 2.x version. We can't not provided the released file for you. You can checkout the source code from https://github.com/ecomfe/echarts/tree/7c36e7dd902dc94b6d3c377acca8784f141ce15e and use it with zrender v2.1.1, Which can be downloaded from https://github.com/ecomfe/zrender/releases.

Thanks for your guys support. We are trying to finish the english doc of ECharts v3 and release it as soom as possible. Word cloud will also been added back again in someway.

**Comment by heremiguel** [comment]
Hi @pissang . Thanks for reply, sorry I don't understand what you mean. Is the url you send me the link to the code fixed or the code base to fix by myself?
Cheers
Jose.

**Comment by pissang** [comment]
It's the code fixed.

**Comment by heremiguel** [comment]
Ok, many thanks @pissang , I will try to locate the file with the fix.
I realised WordCloud is not in the new version 3.x :(

**Comment by pissang** [comment]
Yes, sorry about that. We will migrate the wordCloud to v3 as soon as possible.

**Comment by heremiguel** [comment]
It's fine! little to little.
Regards

**Comment by IevgenySp** [comment]
Hi @pissang. Thanks for your reply and provided solution. Hope WordCloud will be released in ECharts 3.0 soon! :-)

Regards,
Ievgenii

**Comment by pissang** [comment]
https://github.com/ecomfe/echarts-wordcloud
